src/Model/SearchIndexInput.php

```diff
@@ -66,7 +67,7 @@ public function __construct(array $index)
      * @see \MongoDB\Collection::createSearchIndexes()
      * @see https://php.net/mongodb-bson-serializable.bsonserialize
      */
-    public function bsonSerialize(): object
+    public function bsonSerialize(): stdClass
     {
         return (object) $this->index;
```

Comment:
The psalm inferred type is incorrect. `$this->index` is an array, so `(object) $this->index` cannot be anything different that `stdClass`.
Let's add this warnings to the baseline.

> src/Model/SearchIndexInput.php:70:38: MoreSpecificReturnType: The declared return type 'stdClass' for MongoDB\Model\SearchIndexInput::bsonSerialize is more specific than the inferred return type 'object' (see https://psalm.dev/070)

Reply:
Reported issue: https://github.com/vimeo/psalm/issues/10019